dissect/target/filesystems/zip.py

```diff
@@ -85,6 +85,11 @@ def __init__(self, fs: ZipFilesystem, path: str, entry: zipfile.ZipInfo):
         super().__init__(fs, path)
         self.entry = entry
 
+        if self.entry.date_time[0] < 1980 or self.entry.date_time[:3] == (1980, 0, 0):
+            self.entry.date_time = (1980, 1, 1, 0, 0, 0)
+        elif self.entry.date_time[0] > 2107:
+            self.entry.date_time = (2107, 12, 31, 23, 59, 59)
```

Comment:
Can you deal with this in `lstat` instead of here? Would rather keep the original entry untouched.